restaurants/views.py

```diff
@@ -54,13 +55,78 @@ class RestaurantDetailView(DetailView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         restaurant = self.object
+        context['is_owner'] = restaurant.owner == self.request.user
         menu_items = restaurant.menu_items.filter(is_deleted=False)
         self.filterset = self.filterset_class(self.request.GET, queryset=menu_items)
         context['menu_items'] = self.filterset.qs
         context['filter'] = self.filterset
         context['reviews'] = restaurant.reviews.order_by('-id')  
         return context
+
+
+class DishCreateView(CreateView):
+    model = Dish
+    fields = ['name', 'price', 'is_veg', 'image', 'cuisine_type']
+    template_name = 'restaurants/dish_form.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['action'] = 'Add'
+        return context
+
+    def form_valid(self, form):
+        restaurant = get_object_or_404(Restaurant, pk=self.kwargs['restaurant_id'])
+        if restaurant.owner == self.request.user:
+            form.instance.restaurant = restaurant
+            return super().form_valid(form)
+        else:
+            return HttpResponseForbidden()
+        
+    def get_success_url(self):
+        return reverse_lazy('restaurant-detail', kwargs={'pk': self.object.restaurant.pk})
+
+
+class DishUpdateView(UpdateView):
+    model = Dish
+    fields = ['name', 'price', 'is_veg', 'image', 'cuisine_type']
+    template_name = 'restaurants/dish_form.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['action'] = 'Edit'
+        return context
+
+    def get_queryset(self):
+        restaurant = get_object_or_404(Restaurant, pk=self.kwargs['restaurant_id'])
+        if restaurant.owner == self.request.user:
+            return restaurant.menu_items.all()
+        else:
+            return Dish.objects.none()
     
+    def get_success_url(self):
+        return reverse_lazy('restaurant-detail', kwargs={'pk': self.object.restaurant.pk})
+
+class DishDeleteView(DeleteView):
+    model = Dish
+    template_name = 'restaurants/dish_confirm_delete.html'
+
+    def get_queryset(self):
+        restaurant = get_object_or_404(Restaurant, pk=self.kwargs['restaurant_id'])
+        if restaurant.owner == self.request.user:
+            return restaurant.menu_items.all()
+        else:
+            return Dish.objects.none()
```

Comment:
Instead of checking inside view whether the user is the owner or not, you can create a custom decorator `@restauant_owner_required`.

Use that decorator on the dispatch() method of the view. That way if the user does not have permission, they will not be allowed inside the view.

You can use the same decorator on all the admin views.